daemon/container.go

```diff
@@ -1048,6 +1048,16 @@ func (container *Container) HostConfig() *runconfig.HostConfig {
 	return res
 }
 
+func (container *Container) hostConfigWithoutALock() *runconfig.HostConfig {
+	return container.hostConfig
+}
+
+func (c *Container) WriteAs(newContainer *Container) string {
```

Comment:
This **think** was @calavera  was suggesting is that WriteAs() would copy ALL of the config data for a container and populate a "Container" struct that's passed in, not just the NetworkMode property.  Then in the walk func, below, it wouldn't need to lock the container it would just call WriteAs().  Not sure why hostConfigWithoutLock() would be needed though.